utils/build/docker/dotnet_http/parametric/Endpoints/ApmTestApi.cs

```diff
@@ -0,0 +1,325 @@
+    internal static readonly MethodInfo GetAgentWriter = TracerManagerType.GetProperty("AgentWriter", BindingFlags.Instance | BindingFlags.Public)!.GetGetMethod()!;
+    internal static readonly FieldInfo GetStatsAggregator = AgentWriterType.GetField("_statsAggregator", BindingFlags.Instance | BindingFlags.NonPublic)!;
+    private static readonly PropertyInfo SpanContext = SpanType.GetProperty("Context", BindingFlags.Instance | BindingFlags.NonPublic)!;
+    private static readonly PropertyInfo Origin = SpanContextType.GetProperty("Origin", BindingFlags.Instance | BindingFlags.NonPublic)!;
+    private static readonly MethodInfo SetMetric = SpanType.GetMethod("SetMetric", BindingFlags.Instance | BindingFlags.NonPublic)!;
+
+    internal static readonly PropertyInfo SamplingPriority = SpanContextType.GetProperty("SamplingPriority", BindingFlags.Instance | BindingFlags.NonPublic)!;
+    internal static readonly PropertyInfo RawTraceId = SpanContextType.GetProperty("RawTraceId", BindingFlags.Instance | BindingFlags.NonPublic)!;
+    internal static readonly PropertyInfo RawSpanId = SpanContextType.GetProperty("RawSpanId", BindingFlags.Instance | BindingFlags.NonPublic)!;
+    internal static readonly PropertyInfo AdditionalW3CTraceState = SpanContextType.GetProperty("AdditionalW3CTraceState", BindingFlags.Instance | BindingFlags.NonPublic)!;
+
+    // Propagator methods
+    private static readonly MethodInfo SpanContextPropagatorInject = GenerateInjectMethod()!;
+
+    // StatsAggregator flush methods
+    private static readonly MethodInfo StatsAggregatorDisposeAsync = StatsAggregatorType.GetMethod("DisposeAsync", BindingFlags.Instance | BindingFlags.Public)!;
+    private static readonly MethodInfo StatsAggregatorFlush = StatsAggregatorType.GetMethod("Flush", BindingFlags.Instance | BindingFlags.NonPublic)!;
+
+    private static readonly Dictionary<ulong, ISpan> Spans = new();
+
+    internal static ILogger<ApmTestApi> _logger;
+
+    internal static readonly SpanContextExtractor _spanContextExtractor = new();
+
+    internal static IEnumerable<string> GetHeaderValues(string[][] headersList, string key)
+    {
+        List<string> values = new List<string>();
+        foreach (var kvp in headersList)
+        {
+            if (kvp.Length == 2 && string.Equals(key, kvp[0], StringComparison.OrdinalIgnoreCase))
+            {
+                values.Add(kvp[1]);
+            }
+        }
+
+        return values.AsReadOnly();
+    }
+    
+    public static async Task StopTracer()
+    {
+        await Tracer.Instance.ForceFlushAsync();
+    }
+
+    private static async Task<string> StartSpan(HttpRequest request)
+    {
+        var headerRequestBody = await new StreamReader(request.Body).ReadToEndAsync();
+        var parsedDictionary = JsonConvert.DeserializeObject<Dictionary<string, Object>>(headerRequestBody);
```

Comment:
C# style suggestion: Since the DeserializeObject returns a value that may be null, you could try the following so you always have a non-null value assigned to `parsedDictionary`, to avoid having to use the `!` operator later. This will require you to add a using statement to bring in the `System.Collections.Immutable` namespace. Also I recommend changing `Object` to `object` so you're using the built-in alias type

```suggestion
        var parsedDictionary = JsonConvert.DeserializeObject<Dictionary<string, object>>(headerRequestBody) ?? ImmutableDictionary<string, object>.Empty;
```